scene/resources/mesh.cpp

```diff
@@ -201,7 +201,7 @@ Ref<TriangleMesh> Mesh::generate_triangle_mesh() const {
 			continue;
 		}
 		int len = (surface_get_format(i) & ARRAY_FORMAT_INDEX) ? surface_get_array_index_len(i) : surface_get_array_len(i);
-		if ((primitive == PRIMITIVE_TRIANGLES && (len == 0 || (len % 3) != 0)) || (primitive == PRIMITIVE_TRIANGLE_STRIP && len < 3)) {
+		if ((primitive == PRIMITIVE_TRIANGLES && (len == 0 || (len % 3) != 0)) || (primitive == PRIMITIVE_TRIANGLE_STRIP && len < 3) || (surface_get_format(i) & ARRAY_FLAG_USES_EMPTY_VERTEX_ARRAY)) {
```

Comment:
Ultra nitpicky but could we split this over a few more lines? :) Getting unwieldy

Reply:
Split into 3 lines. It isn't in double brackets, each of the 3 primary conditions are each in brackets. The first one because it is a compound condition and the third to avoid order of operation errors with the bit flag

Reply:
D'oh, of course :)